### ISSUE TITLE
How do I create a request interceptor/middleware?

### ISSUE THREAD

**Issue by illunix** [microsoft/kiota-typescript#1492]
How do I create a request interceptor/middleware?

**Comment by baywet** [comment]
Here is an example of one, there are more in the parent directory.

https://github.com/microsoft/kiota-typescript/blob/ecebb900ea828df6851a137f474a7202c474d94f/packages/http/fetch/src/middlewares/retryHandler.ts#L25

Let us know if you have any additional comments or questions.

**Comment by illunix** [comment]
Any idea why it does not include this header in actual request but it shows in console? @baywet 
```ts

  public async execute(
    url: string,
    reqInit: RequestInit,
    reqOptions?: Record<string, RequestOption>
  ): Promise<FetchResponse> {
    return this.executeInternal(url, reqInit as FetchRequestInit);
  }

  private async executeInternal(
    url: string,
    reqInit: FetchRequestInit
  ): Promise<Response> {
    setRequestHeader(reqInit as FetchRequestInit, 'Test', `Test`);

    console.log(reqInit as RequestInit);

    const res = await this.next?.execute(url, reqInit as RequestInit);
    if (!res) {
      throw new Error('Response is undefined');
    }

    return res;
  }
```

![Image](https://github.com/user-attachments/assets/c0f527cd-eaff-4d60-90e0-af48554dec8d)
![Image](https://github.com/user-attachments/assets/7861821b-484f-4f7d-8171-a119d0b85a35)

**Comment by baywet** [comment]
Can you check what versions of the packages are actually installed and upgrade to the latest please? There used to be a a similar bug we fixed in #1358
